From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Fri, 25 Jul 2025 11:56:54 +0000
Subject: [PATCH] restore local backend based password manager settings support

Adapted from GrapheneOS Vanadium patch 0212-restore-local-backend-based-password-manager-setting.patch for the Vandium browser port.

---
 .../browser/password_manager/android/BUILD.gn |  8 +++++
 .../PasswordManagerHelper.java                |  6 ++++
 .../PasswordsPreferenceHelper.java            | 34 +++++++++++++++++++
 .../settings/PasswordsPreference.java         |  4 +++
 ...ssword_manager_settings_service_factory.cc |  3 +-
 5 files changed, 54 insertions(+), 1 deletion(-)
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/PasswordsPreferenceHelper.java

diff --git a/chrome/browser/password_manager/password_manager_settings_service_factory.cc b/chrome/browser/password_manager/password_manager_settings_service_factory.cc
index aa47f805f57d5..6ab7052644dce 100644
--- a/chrome/browser/password_manager/password_manager_settings_service_factory.cc
+++ b/chrome/browser/password_manager/password_manager_settings_service_factory.cc
@@ -10,6 +10,7 @@
 #include "chrome/browser/webid/federated_identity_auto_reauthn_permission_context.h"
 #include "chrome/browser/webid/federated_identity_auto_reauthn_permission_context_factory.h"
 #include "components/password_manager/core/browser/features/password_features.h"
+#include "components/password_manager/core/browser/password_manager_buildflags.h"
 #include "components/password_manager/core/browser/password_manager_settings_service.h"
 #include "components/password_manager/core/browser/password_manager_settings_service_impl.h"
 #include "components/password_manager/core/common/password_manager_features.h"
@@ -83,7 +84,7 @@ PasswordManagerSettingsServiceFactory::BuildServiceInstanceForBrowserContext(
 
 std::unique_ptr<password_manager::PasswordManagerSettingsService>
 PasswordManagerSettingsServiceFactory::CreateService(Profile* profile) const {
-#if BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_ANDROID) && !BUILDFLAG(USE_LOGIN_DATABASE_AS_BACKEND)
   // For the first run after the feature is enabled, before the unmigrated
   // passwords are exported, `IsPasswordManagerAvailable` can return false.
   // However, password saving isn't possible in that run anyway.
